reinforcement_learning/unit_test/live_model_test.cc

```diff
@@ -231,3 +232,87 @@ BOOST_AUTO_TEST_CASE(live_model_mocks) {
   Verify(Method(mock_logger, init)).Exactly(2);
   Verify(Method(mock_logger, append)).Exactly(2);
 }
+
+BOOST_AUTO_TEST_CASE(live_model_logger_receive_data) {
+  Mock<r::i_logger> mock_observation_logger;
+  std::vector<std::string> recorded_observations;
+  When(Method(mock_observation_logger, init)).AlwaysReturn(err::success);
+  When(Method(mock_observation_logger, append)).AlwaysDo(
+    [&recorded_observations](std::string& message, reinforcement_learning::api_status* status) {
+    recorded_observations.push_back(message); return err::success;
+  });
+  Fake(Dtor(mock_observation_logger));
```

Comment:
nit: These mocks are pretty generic. In the future we should try to extract this from the unittest itself.

Reply:
I am going to leave this for another PR though in the interest of getting this one closed out